modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java

```diff
@@ -90,6 +93,9 @@ public ActionRequestValidationException validate() {
         if (destination.timestamp() != null) {
             e = addValidationError("setting timestamp on destination isn't supported. use scripts instead.", e);
         }
+        if (getRemoteInfo() != null && getSearchRequest().source().query() != null) {
+            e = addValidationError("reindex from remote sources should use RemoteInfo's query instead of source's query", e);
```

Comment:
Over REST you can't get this error because I extract the query element from the request before shipping it `SearchRequestSource.parseXContent`. It makes the request **look** clean even though we do this raw query thing.